src/renderer/services/file-storage-service/test/file-storage-service.test.ts

```diff
@@ -49,7 +49,8 @@ describe("FileStorageService", () => {
         file_size: fileSize,
         local_nas_path: undefined,
         local_nas_shortcut: false,
-        multifile: false
+        multifile: false,
+        should_be_in_local: undefined
```

Comment:
Shouldn't `should_be_in_local` always get a boolean value ?

Reply:
should_be_in_local is optional right now and since we are simulating a call we can choose what output we want that to be. We could add should_be_in_local to the required variables and choose to inject the value here. However this doesn't change any of the test logic.

Reply:
Thinking about it a bit more I think it is actually better to have it set as `undefined`. The variable is passed through the registerUpload function. At a higher level we set this value but no changes occur within the registerUpload function itself, by setting it to undefined we can see more easily if some new modification shows up ie defaulting to true or false. Which before would have been invisible.

Reply:
> should_be_in_local is optional right now

Optional in the UI as well as the function? If it's optional in the UI, I agree it should be optional in the function and tested as `undefined` (as well as t/f). However, if it's always defined in the UI, then it seems like it should not be optional in the function or need to be tested as `undefined`

Reply:
Update: it looks like it should always be defined in the UI since it's a checkbox, so I'm not sure I understand your second comment--maybe we can discuss at an FMS standup?

Reply:
I notice that `local_nas_path` is also undefined also.  So I probably am just not understanding this test very well.  But since all the other args are ones that would be valid on the server, I think it would be easier to understand if that was made consistent.

Reply:
This is just testing the FSS call not the data flow from the user's click to the server so it is undefined here because we aren't passing a value explicitly to FSS. IMO it doesn't matter whether you do or don't set it here as long as you assert for the expected value which you do on line 61 & 63

Reply:
I'm afraid I still don't quite understand.

> by setting it to undefined we can see more easily if some new modification shows up ie defaulting to true or false

Is setting `should_be_in_local` to `undefined` a way to guard against any future changes that might change the value somewhere below this function? Is that the idea?

Reply:
yeah its the same way we handle local_nas_path [here](https://github.com/aics-int/aics-file-upload-app/blob/88508b9f2318add7e6c35553e11d9fbd64ff69ad/src/renderer/services/file-storage-service/test/file-storage-service.test.ts#L50)